frontend/TestRun/EventsTab.svelte

```diff
@@ -117,11 +117,29 @@
             return parsed;
         });
 
+        let sortedNemesises = nemesisData.sort((a, b) => a.start_time - b.start_time);
+        let nemesisIndex = 0;
+        for (let event of parsedEvents.sort((a, b) => a.time - b.time)) {
+            event.nemesis = undefined;
+                while(sortedNemesises[nemesisIndex]){
+                    if (event.time.getTime() < sortedNemesises[nemesisIndex].start_time * 1000) {
+                        break;
+                    }else if(event.time.getTime() <= sortedNemesises[nemesisIndex].end_time * 1000){
+                        event.nemesis = sortedNemesises[nemesisIndex];
+                        break;
+                    }else{
```

Comment:
The `else` not needed as we break out in every other conditional, can just `nemesisIndex++` here.